internal/kafka/internal/services/data_plane_cluster_test.go

```diff
@@ -77,6 +77,79 @@ func Test_DataPlaneCluster_UpdateDataPlaneClusterStatus(t *testing.T) {
 				return NewDataPlaneClusterService(sampleValidApplicationConfigForDataPlaneClusterTest(clusterService))
 			},
 		},
+		{
+			name:          "An error is returned when svcErr != nil",
+			clusterID:     testClusterID,
+			clusterStatus: nil,
+			dataPlaneClusterServiceFactory: func() *dataPlaneClusterService {
+				clusterService := &ClusterServiceMock{
+					FindClusterByIDFunc: func(clusterID string) (*api.Cluster, *errors.ServiceError) {
+						return nil, &errors.ServiceError{}
+					},
+				}
+				return NewDataPlaneClusterService(sampleValidApplicationConfigForDataPlaneClusterTest(clusterService))
+			},
+			wantErr: true,
+		},
+		{
+			name:          "An error is returned when Status: api.ClusterFailed ",
```

Comment:
Could you provide where it fails specifically in the `UpdateDataPlaneClusterStatus` method when status is `api.ClusterFailed`?